### PR TITLE
fix safari can not resume the webaudio

### PR DESCRIPTION
**Pull request by tangkaikk**
Re: #17637 

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: pal/audio/web/player-web.ts [hunk 1]
```diff
@@ -95,7 +95,10 @@ export class AudioContextAgent {
                 resolve();
                 return;
             }
-            context.resume().catch((e) => { debug.warn('runContext error', e); });
+            // only state is suspended can involve resume function
+            if (context.state === 'suspended') {
+                context.resume().catch((e) => { debug.warn('runContext error', e); });
+            }
             if (context.state === 'running') {
```
Comment (bofeng-song) [inline pal/audio/web/player-web.ts]: The 'if' condition needs to be changed to 'else if'.

Comment (bofeng-song) [inline pal/audio/web/player-web.ts]: The 'if' condition needs to be changed to 'else if'.

File: pal/audio/web/player-web.ts [hunk 2]
```diff
@@ -95,7 +95,10 @@ export class AudioContextAgent {
                 resolve();
                 return;
             }
-            context.resume().catch((e) => { debug.warn('runContext error', e); });
+            // only state is suspended can involve resume function
+            if (context.state === 'suspended') {
+                context.resume().catch((e) => { debug.warn('runContext error', e); });
```
Comment (bofeng-song) [inline pal/audio/web/player-web.ts:100]: After the logic change, will low-end devices be able to play normally?

File: pal/audio/web/player-web.ts (hunk 1 above)
Comment (tangkaikk) [reply pal/audio/web/player-web.ts]: ok

File: pal/audio/web/player-web.ts (hunk 2 above)
Comment (tangkaikk) [reply pal/audio/web/player-web.ts:100]: I have test ios 15.7 ios 14.8 and ios 17.5.1, three version of iphone.

**Review by bofeng-song — APPROVED** [review]

**Pull request merged**